authorize/grpc.go

```diff
@@ -283,13 +278,21 @@ func getCheckRequestURL(req *envoy_service_auth_v2.CheckRequest) *url.URL {
 		Host:   h.GetHost(),
 	}
 	u.Host = urlutil.GetDomainsForURL(u)[0]
-	// envoy sends the query string as part of the path
-	path := h.GetPath()
-	if idx := strings.Index(path, "?"); idx != -1 {
-		u.Path, u.RawQuery = path[:idx], path[idx+1:]
+
+	// first try using the parse method which properly handles encoding and sets Raw+Path/Query and the regular
+	// Path/Query.
+	if nu, err := u.Parse(h.GetPath()); err == nil {
```

Comment:
Another option...?

We could get the path from the ` X-Original-URL` heaader which -- at least for nginx -- contains the "properly" encoded original URL?

```conf
			proxy_set_header            X-Original-URL          $scheme://$http_host$request_uri;

```

Reply:
What do you think? This would require users setting this in their nginx config, but this is also the default for nginx-ingress.